### DISCUSSION TITLE
Port size determination for the launch pads?

### DISCUSSION THREAD

**Discussion by WorkQuoh** [iqm-finland/KQCircuits#104] [Q&A]
Hi, I was just wondering how the wave port sizes are determined for the launcher pads? For a coplanar waveguide the waveport should be 3*(2g +w), where g is the spacing between the center conductor and the adjacent ground and w is the center conductor's width. The height of the wave port should be 4 times the height of the substrate. 

Do these same rules apply to a waveport placed at the interface of a launch pad? If not how is the size of a waveport determined at the interface of a launch pad?

**Comment by jukkarabina** [comment]
The rules you mention seem to be heuristic. For a coplanar waveguide, the wave port width 3 * (2g + w) sounds quite reasonable as the first guess. Instead, "The height of the wave port should be 4 times the height of the substrate." sounds like a rule for a very specific geometry. In general that doesn't make sense. Perhaps it would be better if the height was roughly the same order of magnitude as the width. 

The same heuristic rules apply if the wave port is placed on the launcher (g is the width of the launcher pad and w is the width of the launcher gap). If you are not interested on the effect of launchers, it's recommended to shrink the simulation box such that you can place the wave ports directly on coplanar waveguides. Take a look on `xs1_full_chip_sim.py`  and try playing with the [`launchers`](https://github.com/iqm-finland/KQCircuits/blob/ddc86d32fecc329e59d8aa57c52cf29e77d7664b/klayout_package/python/scripts/simulations/xs1_full_chip_sim.py#L32) variable to see what I mean.
